Bottom: 69c655994e71af522643f99474add76742e8cc73
Top:    e8223ea4b8860f521aaa28483504d75afbe9a667
Author: Kevin Redon <kevredon@mail.tsaitgaist.info>
Date:   2012-12-07 15:31:58 +0100

From ad2380fdd76dac904c4449f14bb043ed2bc5650c Mon Sep 17 00:00:00 2001
Subject: [PATCH] lines with different clearances are not merged

Problem:
When multiple lines (with same direction, thickness, and clear flags) and moved, they are merged as one line.
Only the clearance of the first line is used for the merged line.
This behaviour is not wished when the lines have different clearances.
Solution:
Verify the lines clearances and only merge lines if they hace the same clearance.

Signed-off-by: Kevin Redon <kevredon@mail.tsaitgaist.info>


---

diff --git a/src/create.c b/src/create.c
index b39ff65..0ee8697 100644
--- a/src/create.c
+++ b/src/create.c
@@ -301,6 +301,7 @@ struct line_info
 {
   Coord X1, X2, Y1, Y2;
   Coord Thickness;
+  Coord Clearance;
   FlagType Flags;
   LineType test, *ans;
   jmp_buf env;
@@ -336,6 +337,8 @@ line_callback (const BoxType * b, void *cl)
     }
   /* remove unnecessary line points */
   if (line->Thickness == i->Thickness
+      /* don't merge lines if the clearances differ  */
+      && line->Clearance == i->Clearance
       /* don't merge lines if the clear flags differ  */
       && TEST_FLAG (CLEARLINEFLAG, line) == TEST_FLAG (CLEARLINEFLAG, i))
     {
@@ -418,6 +421,7 @@ CreateDrawnLineOnLayer (LayerType *Layer,
   info.Y1 = Y1;
   info.Y2 = Y2;
   info.Thickness = Thickness;
+  info.Clearance = Clearance;
   info.Flags = Flags;
   info.test.Thickness = 0;
   info.test.Flags = NoFlags ();
